docs/core/testing/unit-testing-best-practices.md

```diff
@@ -0,0 +1,515 @@
+    };
+
+    var result = glossary.ExclaimAllWords(wordList);
+
+    for(int x = 0; x < result.length; x++)
+    {
+        Assert.Equal(wordList[x] + '!', result[x]);
+    }
+}
+```
+
+#### Better:
+```csharp
+public void ExclaimAllWords_TwoWords_ReturnsArrayOfExclaimedWords()
+{
+    var glossary = new Glossary();
+    var input = new string[] 
+    {
+        "a",
+        "b"
+    };
+    var expected = new string[] 
+    {
+        "a!",
+        "b!"
+    };
+
+    var result = glossary.ExclaimAllWords(input);
+
+    Assert.Equal(expected, result);
+}
+```
+
+### Prefer Helper Methods to Setup and Teardown
+If you require a similar object or state for your tests, prefer a helper method than leveraging Setup and Teardown attributes if they exist.
+
+#### Why?
+- Less confusion when reading the tests since all of the code is visible from within each test.
+- Less chance of setting up too much or too little for the given test.
+- Less chance of sharing state between tests which creates unwanted dependencies between them.
+
+#### Bad:
+```csharp
+Glossary glossary;
+
+[SetUp]
+public void Initialize()
+{
+    glossary = new Glossary();
+}
+
+// more tests..
+
+public void ParseWord_NullValue_ThrowsArgumentException()
+{
+    Assert.Throws<ArgumentException>(() => glossary.ParseWord(null));
+}
+
+public void ParseWord_EmptyString_ReturnsEmptyString()
+{
+    Assert.Empty(glossary.ParseWord(""));
+}
+```
+
+#### Better:
+```csharp
+public void ParseWord_NullValue_ThrowsArgumentException()
+{
+    var glossary = CreateDefaultGlossary();
+
+    var result = () => glossary.ParseWord(null);
+
+    Assert.Throws<ArgumentException>(result);
+}
+
+public void ParseWord_EmptyString_ReturnsEmptyString()
+{
+    var glossary = CreateDefaultGlossary();
+
+    var result = glossary.ParseWord("");
+
+    Assert.Empty(result);
+}
+
+// more tests..
+
+private Glossary CreateDefaultGlossary()
+{
+    return new Glossary();
+}
+```
+
+### Avoid Multiple Asserts
+When writing your tests, try to only include one Assert per test. Common approaches to using only one assert include:
+- Create a separate test for each assert.
+- Use parameterized tests.
+
+#### Why?
+- If one Assert fails, the subsequent Asserts will not be evaluated.
+- Ensures you are not asserting multiple cases in your tests.
+- Gives you the entire picture as to why your tests are failing. 
+
+#### Bad:
+```csharp
+public void IsValidWord_InputIsNullOrEmpty_ReturnsFalse()
+{
+    var glossary = new Glossary();
+
+    Assert.False(glossary.IsValidWord(null)); // potential NullException
+    Assert.False(glossary.IsValidWord("")); // this never gets executed if the above test fails
+}
+```
+
+#### Better:
+```csharp
+[InlineData(null)]
+[InlineData("")]
+public void IsValidWord_InputIsNullOrEmpty_ReturnsFalse(string input)
+{
+    var glossary = new Glossary();
+    
+    var result = glossary.IsValidWord(input);
+
+    Assert.False(result);
+}
+```
+
+### Write Minimally Passing Tests
```

Comment:
I'd recommend moving this up, because it doesn't have either the "Avoid" or "Prefer" qualifiers.